unstract/core/tests/file_storage/test_fs_permanent.py

```diff
@@ -0,0 +1,111 @@
+    READ_TEXT_FILE = os.environ.get("READ_TEXT_FILE")
+    WRITE_PDF_FILE = os.environ.get("WRITE_PDF_FILE")
+    WRITE_TEXT_FILE = os.environ.get("WRITE_TEXT_FILE")
+    TEST_FOLDER = os.environ.get("TEST_FOLDER")
+    GCS_BUCKET = os.environ.get("GCS_BUCKET")
+    TEXT_CONTENT = os.environ.get("TEXT_CONTENT")
+    FILE_STORAGE_ENV = "FILE_STORAGE"
+
+
+def permanent_file_storage(provider: FileStorageProvider):
+    credentials = json.loads(os.environ.get(TEST_CONSTANTS.FILE_STORAGE_ENV))
+    file_storage = PermanentFileStorage(provider=provider, credentials=credentials)
+    assert file_storage is not None
+    return file_storage
+
+
+@pytest.mark.parametrize(
+    "file_storage, file_read_path, read_mode, file_write_path, write_mode",
+    [
+        (
+            permanent_file_storage(provider=FileStorageProvider.GCS),
+            "fsspec-test/input/3.txt",
+            "r",
+            "fsspec-test/output/copy_on_write.txt",
+            "w",
+        )
+    ],
+)
+def test_permanent_fs_copy_on_write(
+    file_storage, file_read_path, read_mode, file_write_path, write_mode
+):
+    if file_storage.exists(file_read_path):
+        file_storage.rm(file_read_path)
+    file_read_contents = file_storage.read(file_read_path, read_mode)
+    print(file_read_contents)
+    if file_storage.exists(file_write_path):
+        file_storage.rm(file_write_path)
+    file_storage.write(file_write_path, write_mode, data=file_read_contents)
+
+    file_write_contents = file_storage.read(file_write_path, read_mode)
+    assert len(file_read_contents) == len(file_write_contents)
+
+
+@pytest.mark.parametrize(
+    "file_storage, file_read_path, read_mode, file_write_path, write_mode",
+    [
+        (
+            permanent_file_storage(provider=FileStorageProvider.Local),
+            "fsspec-test/input/3.txt",
+            "r",
+            "fsspec-test/output/copy_on_write.txt",
+            "w",
+        ),
+    ],
+)
+def test_permanent_fs_copy(
+    file_storage, file_read_path, read_mode, file_write_path, write_mode
+):
+    file_read_contents = file_storage.read(file_read_path, read_mode)
+    print(file_read_contents)
+    if file_storage.exists(file_write_path):
+        file_storage.rm(file_write_path)
+    file_storage.write(file_write_path, write_mode, data=file_read_contents)
+
+    file_write_contents = file_storage.read(file_write_path, read_mode)
+    assert len(file_read_contents) == len(file_write_contents)
+
+
+@pytest.mark.parametrize(
+    "file_storage, from_path, read_mode, to_path, write_mode",
+    [
+        (
+            permanent_file_storage(provider=FileStorageProvider.GCS),
+            "fsspec-test/input/3.txt",
+            "r",
+            "fsspec-test/output/test_write.txt",
+            "w",
+        ),
+    ],
+)
+def test_permanent_fs_download(file_storage, from_path, read_mode, to_path, write_mode):
+    file_read_contents = file_storage.read(from_path, read_mode)
+    print(file_read_contents)
```

Comment:
NIT: Update this to be a debug log message instead and add context on what's being logged